staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest_test.go

```diff
@@ -604,6 +616,64 @@ func TestPatchResourceWithRacingVersionConflict(t *testing.T) {
 	tc.Run(t)
 }
 
+func TestPatchResourceWithInvalidJSONPatch(t *testing.T) {
+	namespace := "bar"
+	name := "foo"
+	uid := types.UID("uid")
+
+	tc := &patchTestCase{
+		name: "TestPatchResourceWithInvalidJSONPatch",
+
+		startingPod: &example.Pod{},
+		updatePod:   &example.Pod{},
+
+		expectedError:     `invalid character 'i' looking for beginning of value`,
+		expectedErrorType: metav1.StatusReasonBadRequest,
+		expectedTries:     1,
+	}
+
+	tc.startingPod.Name = name
+	tc.startingPod.Namespace = namespace
+	tc.startingPod.UID = uid
+	tc.startingPod.ResourceVersion = "2"
+	tc.startingPod.APIVersion = examplev1.SchemeGroupVersion.String()
+	tc.updatePod = tc.startingPod
+
+	// invalid syntax
+	tc.jsonPatch = `invalid json patch`
+
+	tc.Run(t)
+}
+
+func TestPatchResourceWithUnprocessableJSONPatch(t *testing.T) {
+	namespace := "bar"
+	name := "foo"
+	uid := types.UID("uid")
+
+	tc := &patchTestCase{
+		name: "TestPatchResourceWithUnprocessableJSONPatch",
+
+		startingPod: &example.Pod{},
+		updatePod:   &example.Pod{},
+
+		expectedError:     `the server rejected our request due to an error in our request`,
+		expectedErrorType: metav1.StatusReasonInvalid,
+		expectedTries:     1,
+	}
+
+	tc.startingPod.Name = name
+	tc.startingPod.Namespace = namespace
+	tc.startingPod.UID = uid
+	tc.startingPod.ResourceVersion = "2"
+	tc.startingPod.APIVersion = examplev1.SchemeGroupVersion.String()
+	tc.updatePod = tc.startingPod
```

Comment:
I would have made a utility function to return a pod pointer to avoid duplicate code.

Reply:
Done.